app/workers/reactions/bust_homepage_cache_worker.rb

```diff
@@ -0,0 +1,21 @@
+module Reactions
+  class BustHomepageCacheWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :high_priority, retry: 10
+
+    def perform(reaction_id, cache_buster = CacheBuster)
```

Comment:
I think we can remove `CacheBuster` as an argument here and just call it directly down in the method's body.

The tests can be adapted to set expectations directly on `CacheBuster`

Reply:
I'm receiving the following error:
```ruby
1) Reactions::BustHomepageCacheWorker#perform doesn't bust the homepage cache when reactable is a Comment
     Failure/Error: expect(CacheBuster).not_to have_received(:bust)
     
       (CacheBuster).bust(no args)
           expected: 0 times with any arguments
           received: 1 time
     # ./spec/workers/reactions/bust_homepage_cache_worker_spec.rb:23:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:95:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:95:in `block (2 levels) in <top (required)>'
```

It's something related to empty the old cache?

Reply:
It's hard to know without seeing the code, it might be an issue in the job or in the spec

Reply:
The worker is actually this:
```ruby
    def perform(reaction_id)
      reaction = Reaction.find_by(id: reaction_id, reactable_type: "Article")
      return unless reaction&.reactable

      featured_articles_ids = Article.where(featured: true).order("hotness_score DESC").limit(3).pluck(:id)
      return unless featured_articles_ids.include?(reaction.reactable_id)

      reaction.reactable.touch
      CacheBuster.bust("/")
      CacheBuster.bust("/")
      CacheBuster.bust("/?i=i")
      CacheBuster.bust("?i=i")
    end
```

And the spec is like this (that one who is failing):
```ruby
    let(:user) { create(:user) }
    let(:article) { create(:article, featured: true) }
    let(:reaction) { create(:reaction, reactable: article, user: user) }
    let(:comment) { create(:comment, commentable: article) }
    let(:comment_reaction) { create(:reaction, reactable: comment, user: user) }
    let(:worker) { subject }

    before do
      allow(CacheBuster).to receive(:bust)
    end

   it "doesn't bust the homepage cache when reactable is a Comment" do
      worker.perform(comment_reaction.id)
      expect(CacheBuster).not_to have_received(:bust)
    end

```

Reply:
I think that is because of the way I'm calling the `CacheBuster`, what do you think?

Reply:
@luchiago RSpec had problems [in the past](https://github.com/rspec/rspec-mocks/issues/248) with resetting counts between invocations of class methods when using `have_received`. Those should be fixed, but I couldn't get your specs working until I changed them to a more classical AAA (arrange-act-assert) approach:

```ruby
RSpec.describe Reactions::BustHomepageCacheWorker, type: :worker do
  describe "#perform" do
    let(:user) { create(:user) }
    let(:article) { create(:article, featured: true) }
    let(:worker) { subject }

    it "busts the homepage cache when reactable is an Article" do
      reaction = create(:reaction, reactable: article, user: user)
      allow(CacheBuster).to receive(:bust)

      worker.perform(reaction.id)

      expect(CacheBuster).to have_received(:bust).exactly(4)
    end

    it "doesn't bust the homepage cache when reactable is a Comment" do
      comment = create(:comment, commentable: article)
      comment_reaction = create(:reaction, reactable: comment, user: user)
      allow(CacheBuster).to receive(:bust)

      worker.perform(comment_reaction.id)

      expect(CacheBuster).not_to have_received(:bust)
    end

    it "doesn't fail if a reaction doesn't exist" do
      expect do
        worker.perform(Reaction.maximum(:id).to_i + 1)
      end.not_to raise_error
    end
  end
end
```

Personally I prefer this style anyway, since `let` for values that are only used in a single spec isn't particularly useful in my opinion.

Reply:
This worked very well. Thanks for your help!